lib/lev/active_job.rb

```diff
@@ -1,21 +1,42 @@
 module Lev
   module ActiveJob
     class Base < Lev.configuration.job_class
+      attr_accessor(:provider_job_id) unless respond_to?(:provider_job_id)
+
       def self.perform_later(routine_class, *args, &block)
         queue_as routine_class.active_job_queue
-        args.push(routine_class.to_s)
 
-        # Create a new status object and push its ID on to the arguments so that
-        # in `perform` it can be used to retrieve the status when the routine is
-        # initialized.
+        # Create a new status object
         status = Lev::create_status
+
+        # Create a hash containing all the arguments for the routine
+        args_hash = {}
+        args.each_with_index { |arg, index| args_hash[index] = arg }
+
+        # Push the routine class name on to the arguments
+        # so that we can run the correct routine in `perform`
+        args.push(routine_class.to_s)
+
+        # Push the status's ID on to the arguments so that in `perform`
+        # it can be used to retrieve the status when the routine is initialized
         args.push(status.id)
 
+        # Set the job_name and job_args
+        status.set_job_name(routine_class.name)
+        status.set_job_args(args_hash)
```

Comment:
I think we'd rather do this inside each routine for the args we want to track instead of doing it for all args here generically, for three reasons:

1. If we do it inside the routine, we can give the args meaningful names so they can be referenced by name instead of index.  
1. The arguments may not be strings here, but they have to be strings to store them as job args -- since we can't know what they are or how to most meaningfully stringify them, this is probably something best done by the actual routine.
1. It is possible that some routine has an argument that we don't want to track (e.g. it is super long, there's no usefulness in tracking it, etc) -- only the specific routine can know which args we do or do not want to track.

Do you agree?

Here's an example of doing it in a routine: 
 @https://github.com/openstax/tutor-server/blob/66dc7c0be52ebdd82a521011eaf1972e6c542151/app/routines/reassign_published_period_task_plans.rb#L11